lnrpc/invoicesrpc/invoices.proto

```diff
@@ -97,12 +97,29 @@ message AddHoldInvoiceRequest {
 }
 
 message AddHoldInvoiceResp {
+    bytes r_hash = 1;
+
     /*
-    A bare-bones invoice for a payment within the Lightning Network.  With the
+    A bare-bones invoice for a payment within the Lightning Network. With the
     details of the invoice, the sender has all the data necessary to send a
     payment to the recipient.
     */
-    string payment_request = 1;
+    string payment_request = 2;
+
+    /*
+    The "add" index of this invoice. Each newly created invoice will increment
+    this index making it monotonically increasing. Callers to the
+    SubscribeInvoices call can use this to instantly get notified of all added
+    invoices with an add_index greater than this one.
+    */
+    uint64 add_index = 16;
```

Comment:
Why these gaps in the numbers? Also, you can't just re-assign the number 1 from `payment_request` to `r_hash`. That will break backward compatibility on the gRPC level.

Reply:
+1 no need to add this big jump in field number, this proto is unrelated to the add invoice response so can have its own fields.

Reply:
I understand. Did not know it would brake backward compatibility. That explains the gap here: https://github.com/lightningnetwork/lnd/blob/7176a212f76f3539d0c0b89bdc4457e647fdf7c2/lnrpc/rpc.proto#L3094.
Working on a new commit...